src/utils/tables/topic/topicTable.tsx

```diff
@@ -0,0 +1,90 @@
+import Link from "next/link";
+
+import { ConceptLink } from "@/components/molecules/conceptLink/ConceptLink";
+import { QUERY_PARAMS } from "@/constants/queryParams";
+import { IFamilyDocumentTopics, TTableColumn, TTableRow } from "@/types";
+import { firstCase } from "@/utils/text";
+
+export type TTopicTableColumnId = "group" | "concepts";
+
+export const TopicTableColumns: TTableColumn<TTopicTableColumnId>[] = [
+  { id: "group", name: "Group", fraction: 2 },
+  { id: "concepts", name: "Concepts", fraction: 5 },
+];
+
+export type TTopicTableRow = TTableRow<TTopicTableColumnId>;
+
+export const getTopicTableRows = (familyTopics: IFamilyDocumentTopics): TTopicTableRow[] => {
+  const rows: TTopicTableRow[] = [];
+
+  familyTopics.rootConcepts.forEach((rootConcept) => {
+    const hasConcepts = familyTopics.conceptsGrouped[rootConcept.wikibase_id]?.length > 0;
+    if (!hasConcepts) return null;
+
+    rows.push({
+      id: rootConcept.wikibase_id,
+      cells: {
+        group: firstCase(rootConcept.preferred_label),
+        concepts: {
+          label: (
+            <div className="flex flex-wrap gap-y-2 gap-x-1">
+              {familyTopics.conceptsGrouped[rootConcept.wikibase_id]
+                .sort((a, b) => {
+                  const countA = familyTopics.conceptCounts[a.wikibase_id] || 0;
+                  const countB = familyTopics.conceptCounts[b.wikibase_id] || 0;
+                  return countB - countA;
+                })
+                .map((concept, i) => (
+                  <span className="inline-block" key={concept.wikibase_id}>
+                    <ConceptLink
+                      concept={concept}
+                      label={`${firstCase(concept.preferred_label)} (${familyTopics.conceptCounts[concept.wikibase_id] || 0})`}
+                    >
+                      <div className="flex flex-col gap-2">
+                        <h6 className="font-bold">{firstCase(concept.preferred_label)}</h6>
+                        <span className="">{concept.description}</span>
+                        <span className="text-xs text-gray-500">
+                          This concept is mentioned in the following documents, click on the document to view the specific passages:
+                        </span>
+                        <ul className="flex flex-col gap-2">
+                          {/* because the concepts are stored as "wikibase_id:label" we have this weird lookup */}
+                          {familyTopics.documents
```

Comment:
Style critique: This is a lot of TS inside of JSX which is making the surrounding JSX harder to follow. I'd suggest offloading this into a function which can be called in one line.